### ISSUE TITLE
[Bug]: create database case sensitive

### ISSUE THREAD

**Issue by dengn** [matrixorigin/matrixone#4670]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93): 0.5.1
- Hardware parameters: Macbook arm m1 4c16g
- OS type: osx
- Others:
```

### Actual Behavior

mysql> create database tpch;
Query OK, 0 rows affected (0.03 sec)

mysql> use TPCH
ERROR 1049 (42000): Unknown database 'TPCH'

### Expected Behavior

In MySQL

mysql> create database kkkk;
Query OK, 1 row affected (0.05 sec)

mysql> use KKKK;
Database changed
mysql> use kkkk;
Database changed

**Comment by iamlinjunhong** [comment]
Statement of use database is case sensitive, because mysql client will send init db which db name is not origin string command to server. And server has to use `` enclose db name. @dengn

**Comment by iamlinjunhong** [comment]
discussing

**Comment by iamlinjunhong** [comment]
fix after #4172 done

**Comment by fengttt** [comment]
should database name just be case insensitive?

**Comment by iamlinjunhong** [comment]
Can't resolve if use `DB2` is case sensitive. Because mysql protocol is incompatible.

**Comment by iamlinjunhong** [comment]
fixed in 6880d4f6e52ed388648e41f8e6a5e3b0d692cad8

**Comment by dengn** [comment]
Verified in MO. 

mysql> create database tpch;
Query OK, 1 row affected (0.02 sec)

mysql> use TPCH;
Database changed
